### PR TITLE
Activity log: Update rewind dialog

### PR DESCRIPTION
**Pull request by sirreal**
This PR changes the way that rewind confirmation dialogs are displayed.
This is an initial implementation that moves the dialog from a modal into the list of activity items just before the "rewound" item.

see p6TEKc-1C7-p2

Includes #18493 

Todo:
- ~When restore is requested, find next backup point in the future and show the dialog there.~

### CODE REVIEW

File: client/my-sites/stats/activity-log-confirm-dialog/index.jsx [hunk 1]
```diff
@@ -8,98 +9,87 @@ import { localize } from 'i18n-calypso';
...
+					} ) }
+				</a>
+				<HappychatButton
+					className="activity-log-confirm-dialog__more-info-link"
+					href="https://help.vaultpress.com/one-click-restore/"
+				>
+					{ translate( '{{icon /}} Any Questions?', {
+						components: { icon: <Gridicon icon={ 'chat' } /> },
+					} ) }
+				</HappychatButton>
```
Comment (dmsnell) [inline client/my-sites/stats/activity-log-confirm-dialog/index.jsx]: Just noting this in relation to the work @nosolosw has been working on - are we wanting to intentionally further couple a project whose path is divergent from this one?

Comment (sirreal) [reply client/my-sites/stats/activity-log-confirm-dialog/index.jsx]: I'm not sure if your question is directed at @nosolosw or me 🙂 

I'm aware that there is work to move HC to an extension, but I'm not aware of any changes in the way framework components like this should be used. I am happy to make the pertinent changes.

Comment (nosolosw) [reply client/my-sites/stats/activity-log-confirm-dialog/index.jsx]: Thanks for making me aware of this Dennis! I wouldn't want to be a stopping force, so go ahead with using the component. I'm focused now on the underlying tech, so it'll be a few weeks until I can move forward the Happychat extension (or separate npm package) work.

**Comment by eliorivero** [comment]
In ccf5702, updated notice about content deletion:

### Before

captura de pantalla 2017-10-06 a la s 18 01 38
captura de pantalla 2017-10-06 a la s 18 01 13

### After
captura de pantalla 2017-10-06 a la s 17 07 23
captura de pantalla 2017-10-06 a la s 17 06 58

**Comment by eliorivero** [comment]
In fc353f0544 finished making dialog look like mockup: removed unnecessary div inside action wrapper. Separated in two columns in reverse order so secondary actions can be dropped on mobile size. Moved icons outside translate function and wrapped text in span for alignment. Aligned action links and buttons.

### Before
captura de pantalla 2017-10-06 a la s 19 32 12

### After
captura de pantalla 2017-10-06 a la s 20 16 43
captura de pantalla 2017-10-06 a la s 20 16 25

**Comment by eliorivero** [comment]
In https://github.com/Automattic/wp-calypso/pull/18241/commits/b66d78014947d5fffb87621f9e1c44bc054fa297
- introduced rounded corners and more prominent shadow for Rewind dialog
- align buttons toward right when they're stacked and still in second column
- tweak style for icon/text so the latter doesn't wrap around the icon.

captura de pantalla 2017-10-09 a la s 14 15 07

**Pull request merged**